### ISSUE TITLE
Sinon Fake Timers: setInterval() does not work as expected

### ISSUE THREAD

**Issue by fatso83** [sinonjs/lolex#201]
_From @ggsbv on August 28, 2018 16:24_

Sinon's implementation of setInterval() does not work when a string is supplied as the interval.

1. Open Node REPL in the terminal.
2. Do the following and observe the output
```
const sinon = require('sinon')
const clock = sinon.useFakeTimers()
const blah = () => console.log(Date.now())

// passing a string
setInterval(blah, '100')

clock.tick(100)
clock.tick(100)
clock.tick(100)

// passing an int
setInterval(blah, 100)

clock.tick(100)
clock.tick(100)
clock.tick(100)
```

**Expected behavior**
You would expect setInterval to return the intervalID value at each invocation, incrementing by 100 each time. You would expect the same behaviour in both scenarios, regardless of whether the interval value is a string or an integer.

**Screenshots**
![screen shot 2018-08-28 at 17 22 04](https://user-images.githubusercontent.com/20184132/44736358-ed958800-aae6-11e8-9927-ef6f4d909c84.png)

_Copied from original issue: sinonjs/sinon#1888_

**Comment by fatso83** [comment]
_From @mgred on August 30, 2018 14:21_

@ggsbv 
Just a guess... I think this is not related to sinon, but to [lolex](https://github.com/sinonjs/lolex) which is the package in use here. 

The `setInterval` Implentation returns an invocation of [`addTimer`](https://github.com/sinonjs/lolex/blob/master/src/lolex-src.js#L212), which checks if the given delay (timeout) is a finite number `Number.isFinite()`. That returns `false` and delay is set to 0 in case you put in a string value.

I hope that helps in any way...

**Comment by fatso83** [comment]
@ggsbv The entire fix to this issue is just to parse the args before adding the timer. Check out [this line](https://github.com/sinonjs/lolex/blob/master/src/lolex-src.js#L549). Want to submit [your first PR](https://github.com/sinonjs/lolex#running-tests) to lolex? <3<3<3

**Comment by fatso83** [comment]
Fixed by #202
